iquip/apps/builder.py

```diff
@@ -1,31 +1,207 @@
 """App module for editting the build arguments and submitting the experiment."""
 
 import json
-from typing import Any, Callable, Dict, Optional, Tuple
+from typing import Any, Callable, Dict, Optional, Tuple, Union
 
 import requests
 from PyQt5.QtCore import QObject, Qt, QThread, pyqtSignal, pyqtSlot
 from PyQt5.QtWidgets import (
-    QPushButton, QVBoxLayout, QWidget
+    QCheckBox, QComboBox, QDoubleSpinBox, QHBoxLayout, QLabel, QLineEdit, QListWidget,
+    QListWidgetItem, QPushButton, QVBoxLayout, QWidget
 )
 
 import qiwis
 from iquip.protocols import ExperimentInfo
 
+class _BaseEntry(QWidget):
+    """Base class for all argument entries.
+
+    This is a wrapper of each entry.
+    In each subclass, value() must be implemented to return the selected value.
+
+    Attributes:
```

Comment:
The `self.layout` is missing! I think you should explain that it is a `QHBoxLayout` and the label is already added, so the subclasses can just add the input widget (checkbox, spinbox, etc.).

Reply:
Oh, it looks helpful! I will add it.